uxarray/grid/intersections.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+from uxarray.constants import ERROR_TOLERANCE
+from uxarray.grid.utils import cross_fma
+from uxarray.grid.lines import point_within_GCA
+
+
+def get_GCA_GCA_intersection(gca1_cart, gca2_cart):
```

Comment:
There are two types of intersections:

1. Intersection between Great Circle Arcs
2. Intersection between a Great Circle Arc and a Constant Latitude.

That's why I want to distinguish them by putting more details in the naming

Reply:
> Suggest changing the function definition to something like this to simplify it.

There are two types of intersections:

1. Intersection between Great Circle Arcs
2. Intersection between a Great Circle Arc and a Constant Latitude.

That's why I want to distinguish them by putting more details in the naming

Reply:
That makes sense!

I'd keep the name you have right now, but make the `GCA` lowercase.

Reply:
> That makes sense!
> 
> I'd keep the name you have right now, but make the `GCA` lowercase.

GCA is an abbreviation and there's already a released function call `point_within_GCA`

Reply:
I would still make the `gca` lowercase (same in the `point_within_GCA` function) as a stylistic choice.

Reply:
Additionally, we should consider a general "intersection" that would allow for the following operations (this might be slightly out of the scope of this PR, but worth considering)

Would be defined as 
```Python
def gca_intersection(a, b):
```

Where (a) and (b) can be either a:
* Point
* GCA
* Polygon (this one probably needs to be implemented in a future PR) 

What do you think?

Reply:
This idea sounds good in the future.

Just several things need to be noticed:

1. Intersection is not a 1 dimension concept, so I don't think a point should be considered "intersected" with anything. We can use predicates like "is_on" to address this issue

2. Internally, there are only two types of intersections: 1. Intersection between Great Circle Arcs 2. Intersection between a Great Circle Arc and a Constant Latitude.

3. The "intersections" involved with the polygon is still intersections between its edges at the end. 

Therefore I can foresee the unified "intersection" functions is just an assemble of different small intersection functions that deal with different cases

Reply:
Sounds good! Let's get this implementation in and then we can think of linking the functions together with a unified one.